src/ui/components/filters/filteroptionscomponent.js

```diff
@@ -271,6 +281,22 @@ export default class FilterOptionsComponent extends Component {
     if (this.config.storeOnChange) {
       this.apply(this.config.isDynamic);
     }
+
+    if (!this.config.isDynamic) {
+      this.core.globalStorage.on('update', this.name, (data) => {
```

Comment:
I know Clean Code says not to use comments as a crutch, but I think we should have a comment above this block indicating that this listener is fired during back nav. Without a comment, I think it will be unclear to people why this is here. The long term solution is to better architect our storage systems so that this kind of manual reconciliation (from persistentStorage format to globalStorage format) is no longer needed (or we have a better pattern for it). But, that's wayyyyy beyond the scope of this item.

Reply:
Agreeeeed. I updated with a comment!